ledger-keyring-mv2.js

```diff
@@ -0,0 +1,195 @@
+const { BaseLedgerKeyring } = require('./base-ledger-keyring')
+
+const CONNECTION_EVENT = 'ledger-connection-change'
+
+class LedgerKeyringMv2 extends BaseLedgerKeyring {
+  init () {
+    this.iframeLoaded = false
+    this._setupIframe()
+
+    this.currentMessageId = 0
+    this.messageCallbacks = {}
+    this._setupListener()
+
+    return Promise.resolve()
+  }
+
+  destroy () {
+    window.removeEventListener('message', this._eventListener)
+
+    return Promise.resolve()
+  }
+
+  _setupIframe () {
+    this.iframe = document.createElement('iframe')
+    this.iframe.src = this.bridgeUrl
+    this.iframe.allow = `hid 'src'`
+    this.iframe.onload = async () => {
+      // If the ledger live preference was set before the iframe is loaded,
+      // set it after the iframe has loaded
+      this.iframeLoaded = true
+      if (this.delayedPromise) {
+        try {
+          const result = await this.updateTransportMethod(
+            this.delayedPromise.transportType,
+          )
+          this.delayedPromise.resolve(result)
+        } catch (e) {
+          this.delayedPromise.reject(e)
+        } finally {
+          delete this.delayedPromise
+        }
+      }
+    }
+    document.head.appendChild(this.iframe)
+  }
+
+  _setupListener () {
+    this._eventListener = ({ origin, data }) => {
+      if (origin !== this._getOrigin()) {
+        return false
+      }
+
+      if (data) {
+        if (this.messageCallbacks[data.messageId]) {
```

Comment:
```suggestion
        if (data.messageId && this.messageCallbacks[data.messageId]) {
```
*Note: I'm fuzzy on the API here; disregard if guaranteed to exist*

Reply:
nvm, i see now this is verbatim from existing version so out of scope here.